### PR TITLE
Maureen/lp 2390 code review teacher experience

### PR DESCRIPTION
**Pull request by maureensturgeon**
- Fixes teacher experience, so they're able to comment and are unable to close a students review or resolve comments. Previously we were checking the param `viewAsCodeReviewer` to determine if we should the code review experience, but for a teacher this param is not true, so the teacher was seeing the student experience. I've updated it so it checks if the current user is the owner of the code review to determine whether we show the owner or reviewer experience.
- Fixes logic for which tab to show for code_review_v2. Previously, when the param viewingCodeReview was true we were defaulting to the old code review tab even when the code review feature was enabled, this has been fixed.
- Changes the proptype shape for comments, it was still for the code review v1 comment shape, we had forgotten to update it. As a result of it being changed, I needed to update some tests that were using the wrong shape.

## Links

- jira ticket: [LP-2390](https://codedotorg.atlassian.net/browse/LP-2390)

## Testing story
Tested locally, updated unit tests

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineReview.jsx [hunk 1]
```diff
@@ -79,15 +80,15 @@ const CodeReviewTimelineReview = ({
           comments.map(comment => {
             return (
               <Comment
+                isViewingAsCodeReviewOwner={ownerId === currentUserId}
```
Comment (bencodeorg) [inline apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineReview.jsx]: Nit: replace with `viewingAsOwner`?

File: apps/test/unit/templates/instructions/codeReviewV2/CodeReviewTimelineReviewTest.jsx [hunk 2]
```diff
@@ -23,16 +23,18 @@ const DEFAULT_REVIEW = {
   comments: [
     {
       id: 123,
-      commentText: 'Great work on this!',
-      name: 'Steve',
-      timestampString: '2022-03-31T04:58:42.000Z',
+      comment: 'Great work on this!',
```
Comment (bencodeorg) [inline apps/test/unit/templates/instructions/codeReviewV2/CodeReviewTimelineReviewTest.jsx:26]: These comment shape changes are independent of the teacher issue yeah?

**Review by bencodeorg — COMMENTED** [review]
Could you add a little detail to the description about why this fixes the teacher account issue? Does it work if a teacher account is enrolled in a section?

File: apps/test/unit/templates/instructions/codeReviewV2/CodeReviewTimelineReviewTest.jsx (hunk 2 above)
Comment (maureensturgeon) [reply apps/test/unit/templates/instructions/codeReviewV2/CodeReviewTimelineReviewTest.jsx:26]: Oh yes, the proptype shape for the comment was wrong, I was seeing it come up in the console when I was developing, once fixing it, it also needed to be fixed in tests to match the shape.

File: apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineReview.jsx (hunk 1 above)
Comment (maureensturgeon) [reply apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineReview.jsx]: Sure, I can change it!

**Comment by maureensturgeon** [comment]
> Could you add a little detail to the description about why this fixes the teacher account issue? Does it work if a teacher account is enrolled in a section?

I've updated the description to include an explanation of what happened here. The reason I didn't catch this before whileI was developing is that I was logged in as a teacher who is also a levelbuilder, and levelbuilders have extra permissions, which messed things up

**Review by bencodeorg — APPROVED** [review]
LGTM, the PR description updates are very helpful! Maybe add a third bullet about the shape updates, since they account for most of the changes in this PR.

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**